### ISSUE TITLE
Improve price suggestion feedback

### ISSUE THREAD

**Issue by anxolin** [gnosis/dex-react#950]
## Current situation
Price suggestion can be a bit confusing especially when the price estimation is a bit slow.
The current price estimation is done in 2 steps:
- First, it shows an instantaneous first estimate using one unit of the token's price (i.e. price for 1 DAI) multiplied by the sell volume of the user
- Second, it queries the actual price for the sell volume inputed by the user

## The issue
The issue is that the second estimation can take a few seconds and:
- The user might send the transaction before the price estimates finishes (wrong price)
- Price estimation might fail or take too long leaving a not realistic price 
- The user might enter his own price and then, then, when the price estimate is ready it overrides the user's price. It can make the user submit an order using a price that is not what he imputed. Also create a flickering effect.
- It's not clear what is going on

## Proposal
The proposal is to add 3 spinners: one in the receive amount, and one in each price. When we are not 100% sure about the price we show the spinners:
![image](https://user-images.githubusercontent.com/2352112/80017871-84d9fb00-84d5-11ea-8fbe-441df598e418.png)

There's 4 possible things that can happen now:

**[case 1]: Success**
When the price replies with the price, now we are sure about the price, so we show it, and remove the spinners:
![image](https://user-images.githubusercontent.com/2352112/80018130-e0a48400-84d5-11ea-802b-c73826b32c35.png)

**[case 2]: Override**
If BEFORE we get the result of the price, the user modifies the price, we take the price of the user, discard the estimation, and remove the loaders

**[case 3]: Too long**
It's possible that the price estimate takes too much in return his response. Then, we would show a message similar to this:
"The estimation is taking too long. Please [retry], or input your own price."

![image](https://user-images.githubusercontent.com/2352112/80019717-3b3edf80-84d8-11ea-8e2f-74ca6b1103ee.png)

**[case 4]: Error**
Like case 3, but we would show:
"Error getting the estimation. Please [retry], or input your own price."

**Comment by anxolin** [comment]
> The error message has the same case number as case 3.
Fixed, thx

> Why not combine case 3/4 into one and just tell the user that there are issues with the estimation, please retry...We should give him less new to learn as possible.

Yes, it's possible. It's just more informative. None of them should ever happen in the future, so if we ever see one happening, I'd like to know if it failed or it was just slow. The user shouldn't need to learn anything here, these are "edge cases" that we should aim not to see them happening

**Comment by pimato** [comment]
> I'd like to know if it failed or it was just slow

both lead to the same user action "retry" so I would suggest to combine those and then like you said still make sure this should never happen!

**Comment by alfetopito** [comment]
I'm in favour of a slightly different approach.

Instead of automagically replacing the price input once the price estimation is received, display the estimation in a different component.

The component is clickable and on click it'll replace price input value.

This approach has the following benefits in my view:
1. Use input is never replaced without direct user action. User might have set something, but - when price estimation is slow - it's replaced once price estimation is complete
2. Gives more clarity what are the suggested prices
3. Gives clear indication of when the price is loading and when it fails to give an estimation

This is similar, and maybe overlap with, the proposal on https://github.com/gnosis/dex-react/issues/952

**Comment by Velenir** [comment]
I'm also for an on-demand price replacement on button click
Instead of a progressive price replacement in multiple stages that will make already busy TradeWidget even more so

**Comment by anxolin** [comment]
> Instead of automagically replacing the price input once the price estimation is received, display the estimation in a different component.

Maybe you are right, although in my opinion also makes it a bit inconvenient to use.
Right now is very convenient that every time you change the token, swap the tokens, or change the amounts, the price is recalculated and you get the `receivingAmount` automatically.
With your approach, you'll be clicking in the option on every change?

Related to this one, https://github.com/gnosis/dex-react/issues/952 one idea could be that, if the user changes the price by hand, we select and auto generate the "custom" %, this way we have all:
- It autosuggest by default
- works dynamically the calculation of buy tokens
- if the user changes the price, he gets a reference if he is above of below market, and even how much (in the custom percentage)

**Comment by alfetopito** [comment]
Superseded by #1163 and resolved by #1164
